sql/core/src/main/scala/org/apache/spark/sql/Dataset.scala

```diff
@@ -550,8 +551,35 @@ class Dataset[T] private[sql](
       case ExplainMode.Formatted =>
         qe.simpleString(formatted = true)
     }
+  }
+
+  /**
+   * Returns the plan (logical and physical) string with a format specified by a given explain mode.
+   *
+   * @group basic
+   * @since 3.0.0
+   */
+  def toExplainString(mode: String): String = {
+    mode.toLowerCase(Locale.ROOT) match {
+      case "simple" => toExplainString(ExplainMode.Simple)
+      case "extended" => toExplainString(ExplainMode.Extended)
+      case "codegen" => toExplainString(ExplainMode.Codegen)
+      case "cost" => toExplainString(ExplainMode.Cost)
+      case "formatted" => toExplainString(ExplainMode.Formatted)
+      case _ => throw new IllegalArgumentException(s"Unknown explain mode: $mode. Accepted " +
+        "explain modes are 'simple', 'extended', 'codegen', 'cost', 'formatted'.")
+    }
+  }
+
+  /**
+   * Prints the plans (logical and physical) with a format specified by a given explain mode.
+   *
+   * @group basic
+   * @since 3.0.0
+   */
+  def explain(mode: ExplainMode): Unit = {
```

Comment:
Yes, I think there's a similar case at `mode`. How about we have `explain(mode: String)` only instead of `explain(mode: ExplainMode)`? For enum one, I am not sure actually yet (e.g., `joinType`).

Reply:
Aha, I see. So, you mean `ExplainMode` is internally used only?

Reply:
Yes, that was my thinking. WDYT?

Reply:
Yea, to me, the string argument looks more useful than enum (cuz we don't import anything for that). But, we might need more comments about this. cc: @cloud-fan @dongjoon-hyun

Reply:
But `SaveMode` is public. We can have both `explain(String)` and `explain(ExplainMode)`

Reply:
But `joinType` doesn't have enum. `ExplainMode` was added in Spark 3.0 so we don't necessarily expose another API. Actually, isn't using string easier given that explain will be used in a debugging purpose more often?

Reply:
If you guys don't feel strongly, can we just have `explain(String)` alone for now? I somewhat feel a bit strong that we should better start from fewer APIs.

Reply:
yea, I like fewer API designs and I think enum arguments doesn't matter for python/R users.